backend/src/main/java/com/bakdata/conquery/models/jobs/UpdateMatchingStats.java

```diff
@@ -90,17 +90,15 @@ public void calculateConceptMatches(Concept<?> concept, Map<ConceptElementId<?>,
 				Table table = worker.getStorage().getTable(bucket.getImp().getTable());
 
 				for (int event = 0; event < bucket.getNumberOfEvents(); event++) {
-					if (!(concept instanceof TreeConcept)
-						|| cBlock.getMostSpecificChildren() == null
-						|| cBlock.getMostSpecificChildren()[event] == null) {
+					if (!(concept instanceof TreeConcept) || cBlock.getEventMostSpecificChild(event) == null) {
 
 						messages.computeIfAbsent(concept.getId(), (x) -> new MatchingStats.Entry())
 								.addEvent(table, bucket, cBlock, event);
 
 						continue;
 					}
 
-					int[] localIds = cBlock.getMostSpecificChildren()[event];
+					int[] localIds = cBlock.getEventMostSpecificChild(event);
```

Comment:
Du könntest das `cBlock.getEventMostSpecificChild(event)` vor das If packen und das Ergebnis in der Condition benutzen